spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/TaskConfiguration.java

```diff
@@ -181,12 +182,14 @@ public TaskExecutionService taskService(LauncherRepository launcherRepository,
 			TaskAppDeploymentRequestCreator taskAppDeploymentRequestCreator,
 			TaskExplorer taskExplorer,
 			DataflowTaskExecutionDao dataflowTaskExecutionDao,
-			DataflowTaskExecutionMetadataDao dataflowTaskExecutionMetadataDao) {
+			DataflowTaskExecutionMetadataDao dataflowTaskExecutionMetadataDao,
+			@Autowired(required = false) OAuth2TokenUtilsService oauth2TokenUtilsService) {
```

Comment:
In spring5 we can now use `Optional` or even `@Nullable`. I think Optional is preferred way these days as this `required` thing was a spring4 thing.

Reply:
Changed to `@Nullable`